### PR TITLE
TST: Test against pysat v3.0.1RC

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses #(issue)

Tests against pysat v3.0.1RC

Updated - Includes a fix from @jklenzing that originates in #92, fixes (locally) windows file issue. Also pulled in review feedback from @aburrell in #88.

## Type of change

Please delete options that are not relevant.

- pysat v3.0.1 support
- 
# How Has This Been Tested?

Already developed unit tests

**Test Configuration**:
* Operating system: GitHub Actions
* Version number: Python 3.7, 3.8, 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by rstoneback** [comment]
Currently stuck on
```
E                   PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmp6yvi7dw1\\pydineof\\dineof\\test\\dineof_2009-01-01.nc'
46
```

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -12,13 +12,13 @@ jobs:
...
-        python-version: ["3.8", "3.9"]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.8, 3.9]
         numpy_ver: [latest]
         include:
-          - python-version: "3.7"
-            numpy_ver: "1.17"
-            os: "ubuntu-latest"
+          - python-version: 3.7
+            numpy_ver: 1.17
```
Comment (aburrell) [inline .github/workflows/main.yml:20]: The numbers should be in quotes so that yaml always treats them correctly.

**Comment by jklenzing** [comment]
Passes on the pull but not the push.  not sure why

**Comment by rstoneback** [comment]
The windows tests are passing for the 'pull_request' versions but not on the 'push' version.

**Comment by rstoneback** [comment]
The error is actually occurring under the `test_download_warning` for DINEOFs. There shouldn't be any files for it to deal with.

**Comment by rstoneback** [comment]
There is an issue if some read-only files are created. I don't think we are.
https://bugs.python.org/issue26660

**Comment by rstoneback** [comment]
Requesting review from @jklenzing rather than @aburrell since she is currently tied up with other matters.

**Review by jklenzing — APPROVED** [review]
Looks good.  Thanks!

**Pull request merged**